src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -915,15 +926,15 @@ subroutine read_fv3_netcdf_guess(fv3filenamegin)
       allocate( name_metvars3d(GSI_MetGuess_Bundle(it)%n3d))
     end if
     
-    if (laeroana_fv3cmaq) then
+    if (laeroana_fv3cmaq.or.laeroana_fv3smoke) then
       if (.not.allocated(name_chemvars3d)) then
         allocate( name_chemvars3d(GSI_ChemGuess_Bundle(it)%n3d))
```

Comment:
@hongli-wang  could we add a check in an appropriate place on if GSI_chemGuess_bundle(it)%n3d=naero_cmaq_fv3 +naero_smoke_fv3  (with possible additional numbers) ?   I think this will help users avoid possible problems in their setup